stubs/tensorflow/tensorflow/keras/layers.pyi

```diff
@@ -0,0 +1,198 @@
+    # First argument will automatically be cast to layer's compute dtype, but any other tensor arguments will not be.
+    # Also various tensorflow tools/apis can misbehave if they encounter a layer with *args/**kwargs.
+    def __call__(self, inputs: _InputT, *, training: bool = False, mask: _TensorCompatible | None = None) -> _OutputT: ...
+    def call(self, __inputs: _InputT) -> _OutputT: ...
+
+    # input_shape's real type depends on _InputT, but we can't express that without HKT.
+    # For example _InputT tf.Tensor -> tf.TensorShape, _InputT dict[str, tf.Tensor] -> dict[str, tf.TensorShape].
+    def build(self, input_shape: Any) -> None: ...
+    @overload
+    def compute_output_shape(self: Layer[tf.Tensor, tf.Tensor], input_shape: tf.TensorShape) -> tf.TensorShape: ...
+    @overload
+    def compute_output_shape(self, input_shape: Any) -> Any: ...
+    def add_weight(
+        self,
+        name: str | None = None,
+        shape: Iterable[int | None] | None = None,
+        dtype: tf._DTypeLike | None = None,
+        initializer: _Initializer | None = None,
+        regularizer: _Regularizer = None,
+        trainable: bool | None = None,
+        constraint: _Constraint = None,
+        use_resource: bool | None = None,
+        synchronization: VariableSynchronization = ...,
+        aggregation: VariableAggregation = ...,
+    ) -> tf.Variable: ...
+    def add_loss(self, losses: tf.Tensor | Sequence[tf.Tensor] | Callable[[], tf.Tensor]) -> None: ...
+    def count_params(self) -> int: ...
+    @property
+    def trainable_variables(self) -> list[Variable]: ...
+    @property
+    def non_trainable_variables(self) -> list[Variable]: ...
+    @property
+    def trainable_weights(self) -> list[Variable]: ...
+    @property
+    def non_trainable_weights(self) -> list[Variable]: ...
+    @property
+    def losses(self) -> list[Tensor]: ...
+    def get_weights(self) -> list[_AnyArray]: ...
+    def set_weights(self, weights: Sequence[_AnyArray]) -> None: ...
+    def get_config(self) -> dict[str, Any]: ...
+    @classmethod
+    def from_config(cls, config: dict[str, Any]) -> Self: ...
+    def __getattr__(self, name: str) -> Incomplete: ...
+
+# Every layer has trainable, dtype, name, and dynamic. At runtime these
+# are mainly handled with **kwargs, passed up and then validated.
+# In actual implementation there's 12 allowed keyword arguments, but only
+# 4 are documented and other 8 are mainly internal. The other 8 can be found
+# https://github.com/keras-team/keras/blob/e6784e4302c7b8cd116b74a784f4b78d60e83c26/keras/engine/base_layer.py#L329
+# PEP 692 support would be very helpful here and allow removing stubtest allowlist for
+# all layer constructors.
+
+# TODO: Replace last Any after adding tf.keras.mixed_precision.Policy.
+_LayerDtype = tf._DTypeLike | dict[str, Any] | Any
+
+# Layer's compute_output_shape commonly have instance as first argument name instead of self.
```

Comment:
I'm unsure it's worthwhile to actually have `def compute_output_shape(instance, input_shape: tf.TensorShape) -> tf.TensorShape: ...` for layers where stubtest complains about instance vs self. Feels very implementation detail and maybe it be better to just allowlist all build/compute_output_shape for layer's.

Reply:
I would go with the simplest solution that passes our CI while maintaining safety. Consider marking the parameter as positional-only, for example.

Reply:
Positional only looks to make ci happy so I’ve done that.